test/python_utils/gitdist_UnitTests.py

```diff
@@ -196,6 +200,112 @@ def test_row_mismatch(self):
+      "| ID | Repo Dir                       | Branch                  | Tracking Branch                          | C | M | ? |\n" \
+      "|----|--------------------------------|-------------------------|------------------------------------------|---|---|---|\n" \
+      "|  0 | MockProjectDir (Base)          | medium_branch           | medium_remote/medium_branch              |   |   |   |\n" \
+      "|  1 | ExtraRepo1                     | short_br                | origin/short_br                          |   |   |   |\n" \
+      "|  2 | really/long/path/to/ExtraRepo2 | really_long_branch_name | long_remote_name/really_long_branch_name |   |   |   |\n" \
+      "------------------------------------------------------------------------------------------------------------------------\n"
+    self.assertEqual(asciiTable, asciiTable_expected)
+
+
+  def test_stty_size_slightly_too_small(self):
+    os.environ["GITDIST_UNIT_TEST_STTY_SIZE"] = "60 106"
+    asciiTable = createAsciiTable(self.create_stty_size_table())
+    #print(asciiTable)
+    asciiTable_expected = \
+      "----------------------------------------------------------------------------------------------------------\n" \
+      "| ID | Repo Dir                  | Branch              | Tracking Branch                     | C | M | ? |\n" \
+      "|----|---------------------------|---------------------|-------------------------------------|---|---|---|\n" \
+      "|  0 | MockProjectDir (Base)     | medium_branch       | medium_remote/medium_branch         |   |   |   |\n" \
+      "|  1 | ExtraRepo1                | short_br            | origin/short_br                     |   |   |   |\n" \
+      "|  2 | really/long.../ExtraRepo2 | really_l...nch_name | long_remote_name...long_branch_name |   |   |   |\n" \
+      "----------------------------------------------------------------------------------------------------------\n"
+    self.assertEqual(asciiTable, asciiTable_expected)
+
+
+  def test_stty_size_significantly_too_small(self):
+    os.environ["GITDIST_UNIT_TEST_STTY_SIZE"] = "60 70"
+    asciiTable = createAsciiTable(self.create_stty_size_table())
+    #print(asciiTable)
+    asciiTable_expected = \
+      "----------------------------------------------------------------------\n" \
+      "| ID | Repo Dir      | Branch     | Tracking Branch      | C | M | ? |\n" \
+      "|----|---------------|------------|----------------------|---|---|---|\n" \
+      "|  0 | MockP...Base) | medi...nch | medium_re...m_branch |   |   |   |\n" \
+      "|  1 | ExtraRepo1    | short_br   | origin/short_br      |   |   |   |\n" \
+      "|  2 | reall...Repo2 | real...ame | long_remo...nch_name |   |   |   |\n" \
+      "----------------------------------------------------------------------\n"
+    self.assertEqual(asciiTable, asciiTable_expected)
+
+
+  def test_stty_size_too_small_for_heading(self):
```

Comment:
Okay, and here is the test for the behavior when the terminal width is too small to fit all of the headers and it prints the full table.  Nice!